src/shared/service-util.c

```diff
@@ -24,12 +24,14 @@ static int help(const char *program_path, const char *service, const char *descr
                "  -h --help                 Show this help\n"
                "     --version              Show package version\n"
                "     --bus-introspect=PATH  Write D-Bus XML introspection data\n"
-               "\nSee the %s for details.\n"
-               , program_path
-               , ansi_highlight(), description, ansi_normal()
-               , ansi_underline(), ansi_normal()
-               , link
-        );
+               "\nSee the %s for details.\n",
+               program_path,
+               ansi_highlight(),
+               description,
+               ansi_normal(),
```

Comment:
i kinda link if the start ansi sequence, the covered string and the stop ansi sequence are put together on one line. but doesn't matter. i.e. putting these three one one line is kinda nice i think

Reply:
those are two very different cases. one uses positional referencing in the format expression (i.e. `%5$s` and similar) and the other one uses automatic referencing (i.e. `%s` and similar). If we use positional referencing then the argument list does not match the order how things appear in the output, hence you cannot sensibly arrange things so that the "sequence covers string x" logic actually can be made visible via line breaks.

Reply:
I think you're checking only the first commit, the second one added positional referencing, so both cases are identical:

https://github.com/systemd/systemd/pull/31721/files#diff-50360e0b753c9ffcc8d21c5579a7564de0af85862eb3e977977a379aa05c2444R21

vs

https://github.com/systemd/systemd/blob/14300cdde437bdc8aa0fe309f84c9c532ef016e6/src/sysext/sysext.c#L1989

Reply:
oh, uh, sorry. ignore me then ;-)

Reply:
no problem, thanks :)